### PR TITLE
Examples: Using CARGO UDC in rust example

### PR DESCRIPTION
**Pull request by adamgordonbell**
Rust example uses our UDC instead of cargo chef

### CODE REVIEW

File: examples/rust/Earthfile [hunk 1]
```diff
@@ -1,29 +1,19 @@
-VERSION 0.7
-FROM rust:1.72
-WORKDIR /rustexample
-
-install-chef:
-    RUN cargo install --debug cargo-chef
+VERSION --global-cache 0.7
+IMPORT github.com/earthly/lib/rust AS rust
```
Comment (vladaionescu) [inline examples/rust/Earthfile:2]: It might be good practice to pin this to a tag. It'll help with not breaking people's setup when we change the UDC.

**Review by vladaionescu — COMMENTED** [review]
Leaving the main review to Nacho

File: examples/rust/Earthfile [hunk 2]
```diff
@@ -1,29 +1,19 @@
...
-# Using cutoff-optimization to ensure cache hit (see examples/cutoff-optimization)
-build-cache:
-    FROM +install-chef
-    COPY +prepare-cache/recipe.json ./
-    CACHE target
-    RUN cargo chef cook --release
+FROM rust:slim-buster
+WORKDIR /rustexample
 
+# Cache dependencies
```
Comment (idodod) [inline examples/rust/Earthfile:7]: would be nice to use comments that work with our earthly doc command.
e.g. 
```suggestion
# build creates a docker image using cached rust commands
```

Comment (adamgordonbell) [reply examples/rust/Earthfile:7]: oh yeah. Good idea

File: examples/rust/Earthfile [hunk 3]
```diff
@@ -1,29 +1,19 @@
...
-# Using cutoff-optimization to ensure cache hit (see examples/cutoff-optimization)
-build-cache:
-    FROM +install-chef
-    COPY +prepare-cache/recipe.json ./
-    CACHE target
-    RUN cargo chef cook --release
+FROM rust:slim-buster
+WORKDIR /rustexample
 
+# build creates a docker image using cached rust commands
```
Comment (idodod) [inline examples/rust/Earthfile:7]: ooh oops, it doesn't actually creates a docker image :)

Comment (adamgordonbell) [reply examples/rust/Earthfile]: Details matter 🤣

**Review by idelvall — APPROVED** [review]

File: examples/rust/Earthfile [hunk 4]
```diff
@@ -1,29 +1,19 @@
...
-    COPY --dir src Cargo.lock Cargo.toml .
-    RUN cargo build --release --bin example-rust
+    # Cargo UDC adds caching to cargo runs.
+    # See https://github.com/earthly/lib/tree/main/rust
+    DO rust+INIT --keep_fingerprints=true
+    COPY --keep-ts --dir src Cargo.lock Cargo.toml .
+    DO rust+CARGO --args="build --release --bin example-rust" --output="release/[^/\.]+"
     SAVE ARTIFACT target/release/example-rust example-rust
 
+# docker creates docker image earthly/examples:rust
```
Comment (vladaionescu) [inline examples/rust/Earthfile:16]: This comment in inaccurate - can you use `SAVE IMAGE --push` here?

Comment (vladaionescu) [reply examples/rust/Earthfile:16]: Ignore me - I missed that

**Pull request merged**